src/eth_provider/provider.rs

```diff
@@ -165,6 +169,8 @@ pub trait EthereumProvider {
     async fn txpool_transactions(&self) -> EthProviderResult<Vec<Transaction>>;
     /// Returns the content of the pending pool.
     async fn txpool_content(&self) -> EthProviderResult<TxpoolContent>;
+    /// Remove the signer cached balance if the transaction has been removed from the pending pool
+    async fn remove_from_account_balances(&self, signer: Address);
```

Comment:
I'm looking for a solution to remove this method from the `EthProvider` trait, but still make it accessible from outside.
@greged93 proposed these two solutions:

- Add a trait called `CachedEthProvider`, which is a super trait of `EthProvider`.
- Add a structure `Cache<P=EthProvider>`.

What do you think, @tcoratger ?